Disco l475 iot1 wifi

VVESTM opened the pull request:
Hi,

Here is a first step to support wifi on disco_l475_iot board.
It provides the library corresponding to the embedded wifi chip. (Inventek ism43362-m3g-l44)

Next : integrate this with control and data plans.

To test this PR, I have used sample applications that directly called the library code. They can be found in my local repo : https://github.com/VVESTM/zephyr/tree/disco_l475_iot1_wifi_samples

Regards,
Vincent

VVESTM committed:
ext: hal: add Inventek ISM43362-M3G-L44 wifi library

This library is to be used with Inventek ism43362-m3g-l44 wifi chip.
It is coming from STMicrolectronics cube examples
(Projects\B-L475E-IOT01\Applications\WiFi\Common directory from
STM32Cube_FW_L4_V1.10.0)

Signed-off-by: Vincent Veron <vincent.veron@st.com>

erwango commented on ext/hal/inventek/ism43362-m3g-l44/Kconfig:
```diff
@@ -0,0 +1,30 @@
+# Kconfig - Inventek wifi hal config
+
+# Copyright (c) 2016 Linaro Limited.
```
Use ST copyright here

erwango commented on ext/hal/inventek/ism43362-m3g-l44/CMakeLists.txt:
```diff
@@ -0,0 +1,16 @@
+# Inventek ism43362-m3g-l44 WiFi chip
+#
+# Copyright (c) 2016 Linaro Limited
```
ST Copyright

erwango commented on ext/hal/inventek/ism43362-m3g-l44/README:
```diff
@@ -0,0 +1,44 @@
+INVENTEK ISM43362-M3G-L44
```
Move in initial commit

erwango commented on ext/hal/inventek/ism43362-m3g-l44/Kconfig:
```diff
@@ -0,0 +1,30 @@
+config WIFI_SPI_DEV_NAME
+	string
+	prompt "SPI used for wifi"
+	default "SPI_X"
```
Set a valid value

erwango commented on ext/hal/inventek/ism43362-m3g-l44/Src/es_wifi_io.c:
```diff
@@ -8,123 +8,99 @@
+void WIFI_RESET_MODULE(void)
+{
+  struct device *dev;
+  dev = device_get_binding("GPIOE");
```
Define and use global variable

erwango commented on ext/hal/inventek/ism43362-m3g-l44/Src/es_wifi_io.c:
```diff
@@ -8,123 +8,99 @@
+void WIFI_ENABLE_NSS(void)
+{
+  struct device *dev;
+  dev = device_get_binding("GPIOE");
```
ditto

erwango commented on ext/hal/inventek/ism43362-m3g-l44/Src/es_wifi_io.c:
```diff
@@ -8,123 +8,99 @@
+{
+  struct device *dev;
+  dev = device_get_binding("GPIOE");
+  gpio_pin_configure(dev, 0, GPIO_DIR_OUT);
```
ditto

erwango reviewed (changes requested):
Some comments.
I think that library porting to zephyr will eventually take place under driver wifi/inventek_private.c. Even if this would be an empty shell waiting the next steps of wifi management. It could be interesting to start it here. @galak, @tbursztyka : What is your opinion on this?

tbursztyka commented on ext/hal/inventek/ism43362-m3g-l44/Inc/es_wifi_conf.h:
```diff
@@ -6,31 +6,31 @@
   ******************************************************************************
```
I'd rather see all this adaptation below in drivers/wifi once you'll be implementing the actual wifi driver.

The idea being: if you need to upgrade the ext/hal, it will be painful to backport your modifications. So, imo, the least amount of modifications should be done in ext/hal, rest of the adaptation in the relevant native part (drivers/, ...).

See what I did here in PR #4711 (original PR was mixing too much modifications in drivers and ext/hal), in the end besides the CMakeLists.txt, I only had to modify one unique file from the whole original ones from ASF.

tbursztyka commented on ext/hal/inventek/ism43362-m3g-l44/Kconfig:
```diff
@@ -0,0 +1,30 @@
+config HAS_INVENTEK_WIFI
+    bool
+
+config WIFI_SPI_DEV_NAME
```
Any Zephyr specific part for adapting the HAL to work, as noticed in other commit, should go - if possible - in drivers/wifi etc... It will then centralize all the configuration required to get the driver working.

VVESTM commented:
I pushed a new version including typo and quick fixes from @erwango (copyrights, ...)

For the customization part of the library, if you agree, I will add an inventek_private.c / .h in driver/wifi that will contain the equivalent of ext/hal/inventek/ism43362-m3g-l44/Src/es_wifi_io.c

I have to check how to deal with Licences. For now, it is BSD 3 clauses and I guess you will want Apache 2.0...

I didn't do it at the beginning because I have no driver yet... My plan was to store it here for now and move to driver once we have control and data plans.

VVESTM committed:
drivers: wifi : add Inventek customization

This is the customization needed by ext/hal/inventek/ism43362-m3g-l44
to run on Zephyr.
It mainly concern spi and gpio.

Signed-off-by: Vincent Veron <vincent.veron@st.com>

VVESTM committed:
driver: wifi: enable inventek wifi compilation

Provide all the necessary files to compile the ext hal and the driver
for Inventek wifi component.

Signed-off-by: Vincent Veron <vincent.veron@st.com>

VVESTM commented:
Hi,
Here is a new version with the customization of the lib done in a wifi driver. I used the same approach as @tbursztyka for the Kconfig.

There is still some licence issue, but your feedback is welcome.

pfalcon commented:
@erwango, @VVESTM : Ping about this PR. Now that there's finally a WiFi driver available in the mainline, maybe some useful updates can be made here?

galak commented:
closing since we now have #10464
